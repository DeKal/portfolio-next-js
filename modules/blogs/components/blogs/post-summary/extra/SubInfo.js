import styled from 'styled-components/macro'

export default styled.div`
  font-weight: 300;
  font-size: ${props => props.theme.fontSizes.xsmall};
  color: ${props => props.theme.colors.lightDark};
`
