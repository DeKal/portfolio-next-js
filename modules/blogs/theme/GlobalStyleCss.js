import React from 'react'
import PropTypes from 'prop-types'

const GlobalStyleCss = () => {
  return (
    <style jsx global>{`
      p {
        margin-top: 1.5em;
        margin-bottom: 5px;
      }
      h1 {
        margin-top: 2em;
      }
      h2,
      h3,
      h4,
      h5,
      h6 {
        margin-top: 1.7em;
      }
    `}</style>
  )
}

GlobalStyleCss.propTypes = {
  theme: PropTypes.any
}

export default GlobalStyleCss
