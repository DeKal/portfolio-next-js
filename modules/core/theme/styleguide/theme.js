import baseTheme from './baseTheme'

const theme = {
  ...baseTheme,
  navHeaderHeight: '40px',
  navContentMinHeight: '80px',
  footerHeight: '50px',
  navHeightWhenScrollingOnMobile: '45px',
  navPosWhenHiding: '-220px',
  aboutMarginTop: '-200px',
  colors: {
    // Color variables
    ghostWhite: '#f8f8ff ',
    colorLight: '#ecf0f1',
    colorDarken: '#bdc3c7',
    colorDark: '#2c3e50',
    colorAccent: '#FFA726',
    colorBlack: '#000000',
    bgColor: '#fff',
    grey: '#696969'
  },
  images: {
    background: '/images/bg.webp',
    backgroundMobile: '/images/bg-mobile.webp',
    bgPattern: '/images/bg-pattern.jpg',
    bgPatternOpacity: '1',
    overlayBgColor: 'rgba(44, 62, 80, 0.4)',
    overlayBgColorMobile: 'rgba(44, 62, 80, 0.3)',
    overlay:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC'
  },
  productImageRatio: '143%',
  headerPadding: '200px 10px 250px'
}

const themeExtra = {
  ...theme,
  extra: {
    boxShadow: `0 1px 2px -1px ${theme.colors.colorDark}`
  }
}

export default themeExtra
