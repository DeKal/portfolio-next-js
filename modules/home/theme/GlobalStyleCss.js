import React from 'react'
import PropTypes from 'prop-types'

const GlobalStyleCss = ({ theme }) => {
  return (
    <style jsx global>{`
      html,
      body {
        color: ${theme.colors.colorDark};
      }
      h1 {
        color: ${theme.colors.colorLight};
      }
      a {
        color: ${theme.colors.colorDark};
        border-bottom: 1px dashed ${theme.colors.colorDark};
      }
      body:after {
        content: '';
        position: fixed;
        top: 0;
        bottom: 0;
        width: 100%;
        height: 100%;
        opacity: ${theme.images.bgPatternOpacity};
        z-index: -1;
        background: ${theme.colors.bgColor} url(${theme.images.bgPattern});
      }
      *:selection {
        color: ${theme.colors.colorLight};
        background-color: ${theme.colors.colorDarken};
      }
      *::-moz-selection {
        color: ${theme.colors.colorLight};
        background-color: ${theme.colors.colorDarken};
      }

      @font-face {
        font-family: 'ip';
        src: url('/fonts/ip/font/ip.eot');
        src: url('/fonts/ip/font/ip.eot#iefix') format('embedded-opentype'),
          url('/fonts/ip/font/ip.woff') format('woff'),
          url('/fonts/ip/font/ip.ttf') format('truetype'),
          url('/fonts/ip/font/ip.svg') format('svg');
        font-weight: normal;
        font-style: normal;
      }

      html,
      body,
      div,
      span,
      applet,
      object,
      iframe,
      h1,
      h2,
      h3,
      h4,
      h5,
      h6,
      p,
      blockquote,
      pre,
      a,
      abbr,
      acronym,
      address,
      big,
      cite,
      code,
      del,
      dfn,
      em,
      img,
      ins,
      kbd,
      q,
      s,
      samp,
      small,
      strike,
      strong,
      sub,
      sup,
      tt,
      var,
      b,
      u,
      i,
      center,
      dl,
      dt,
      dd,
      ol,
      ul,
      li,
      fieldset,
      form,
      input,
      label,
      legend,
      table,
      caption,
      tbody,
      tfoot,
      thead,
      tr,
      th,
      td,
      article,
      aside,
      canvas,
      details,
      embed,
      figure,
      figcaption,
      footer,
      header,
      hgroup,
      menu,
      nav,
      output,
      ruby,
      section,
      summary,
      time,
      mark,
      audio,
      video {
        margin: 0;
        padding: 0;
        border: 0;
        -moz-box-sizing: border-box;
        -webkit-box-sizing: border-box;
        box-sizing: border-box;
      }

      article,
      aside,
      details,
      figcaption,
      figure,
      footer,
      header,
      hgroup,
      menu,
      nav,
      section {
        display: block;
      }

      nav ol,
      nav ul {
        list-style: none;
      }

      blockquote,
      q {
        quotes: none;
      }

      blockquote cite {
        font-style: normal;
      }

      blockquote:before,
      blockquote:after,
      q:before,
      q:after {
        content: '';
        content: none;
      }

      table {
        border-collapse: collapse;
        border-spacing: 0;
      }

      button,
      input[type='submit'] {
        cursor: pointer;
      }

      html,
      body {
        width: 100%;
        height: 100%;
        font-family: 'open sans', Tahoma, Verdana, Arial, sans-serif;
        line-height: 1.2;
        scroll-behavior: smooth;
      }

      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        font-weight: normal;
      }

      h1 {
        font-size: 60px;
        font-family: Georgia, Times, 'Times New Roman', serif;
      }
      @media screen and (max-width: 1200px) {
        h1 {
          font-size: 55px;
        }
      }
      @media screen and (max-width: 800px) {
        h1 {
          font-size: 50px;
        }
      }

      h2 {
        font-family: Georgia, Times, 'Times New Roman', serif;
        font-size: 40px;
      }
      @media screen and (max-width: 1200px) {
        h2 {
          font-size: 35px;
        }
      }
      @media screen and (max-width: 800px) {
        h2 {
          font-size: 30px;
        }
      }

      h3 {
        font-size: 30px;
      }
      @media screen and (max-width: 1200px) {
        h3 {
          font-size: 25px;
        }
      }
      @media screen and (max-width: 800px) {
        h3 {
          font-size: 20px;
        }
      }

      h4 {
        font-size: 20px;
      }
      @media screen and (max-width: 1200px) {
        h4 {
          font-size: 18px;
        }
      }

      h5 {
        font-size: 18px;
      }
      @media screen and (max-width: 800px) {
        h5 {
          font-size: 16px;
        }
      }

      p {
        font-size: 16px;
        line-height: 1.5;
      }
      @media screen and (max-width: 800px) {
        p {
          font-size: 15px;
        }
      }

      li p {
        font-family: Georgia, Times, 'Times New Roman', serif;
        font-style: italic;
      }

      a {
        font-family: 'open sans', Tahoma, Verdana, Arial, sans-serif;
        font-size: 16px;
        text-decoration: none;
        outline-style: none;
      }
      a:hover {
        border-bottom: none;
      }
    `}</style>
  )
}

GlobalStyleCss.propTypes = {
  theme: PropTypes.any
}

export default GlobalStyleCss
