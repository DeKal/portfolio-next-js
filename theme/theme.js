const theme = {
  font:
    '-apple-system, blinkmacsystemfont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
  navHeaderHeight: '40px',
  navContentMinHeight: '80px',
  footerHeight: '50px',
  navHeightWhenScrollingOnMobile: '45px',
  navPosWhenHiding: '-220px',
  aboutMarginTop: '-200px',
  colors: {
    // Color variables
    colorLight: '#ecf0f1',
    colorDarken: '#bdc3c7',
    colorDark: '#2c3e50',
    colorAccent: '#FFA726',
    bgColor: '#fff'
  },
  spaces: {
    xxxsmall: 7,
    xxsmall: 10,
    xsmall: 13,
    small: 16,
    medium: 25,
    large: 35,
    xlarge: 45,
    gutter: 16
  },
  fonts: {
    fontSans: '"Open Sans", Tahoma, Verdana, Arial, sans-serif',
    fontSerif: 'Georgia, Times, "Times New Roman", serif',
    fontCondensed: '"Open Sans Condensed", Tahoma, Verdana, Arial, sans-serif',
    fontIp: '"ip", sans-serif',
    fontTrebuchet: 'Trebuchet MS, sans-serif',
    fontCert: 'Times, serif'
  },
  fontSizes: {
    base: '14px',
    XS: '10px',
    S: '12px',
    M: '14px',
    L: '16px',
    XL: '18px',
    XXL: '22px',
    XXXL: '24px'
  },
  images: {
    background: '/images/bg.jpg',
    bgPattern: '/images/bg-pattern.jpg',
    bgPatternOpacity: '1',
    overlayBgColor: 'rgba(44, 62, 80, 0.3)',
    overlay:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC'
  },
  productImageRatio: '143%',
  defaultLineHeight: 1.57,
  icons: {
    goTop: '\\e817',
    location: '\\e806',
    mail: '\\e800',
    phone: '\\e81a',
    fb: '\\e811',
    linkedin: '\\e815',
    workexp: '\\e804',
    link: '\\e819',
    c: 'C',
    p: 'P'
  },
  headerPadding: '200px 10px 400px'
}

const themeExtra = {
  ...theme,
  extra: {
    boxShadow: `0 1px 2px -1px ${theme.colors.colorDark}`
  }
}

export default themeExtra
