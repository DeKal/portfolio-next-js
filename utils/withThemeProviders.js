import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
