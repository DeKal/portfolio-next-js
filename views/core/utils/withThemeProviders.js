import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
