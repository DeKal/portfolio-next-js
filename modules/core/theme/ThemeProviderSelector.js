import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import {
  isChristmasDay,
  isNewYearComing,
  isTet,
  isBlogs
} from '~/modules/core/utils/helpers'
import theme from '~/modules/core/theme/styleguide/theme'
import christMasTheme from '~/modules/core/theme/styleguide/christMasTheme'
import newYearTheme from '~/modules/core/theme/styleguide/newYearTheme'
import tetTheme from '~/modules/core/theme/styleguide/tetTheme'
import blogTheme from '~/modules/core/theme/styleguide/blogTheme'
import BlogGlobalStyleCss from '~/modules/post/theme/GlobalStyleCss'
import HomeGlobalStyleCss from '~/modules/home/theme/GlobalStyleCss'
import { useRouter } from 'next/router'

const ThemeProviderSelector = ({ children }) => {
  const router = useRouter()
  const theme = useTheme(router)
  return (
    <ThemeProvider theme={theme}>
      {children}
      {selectGlobalCss(router, theme)}
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any
}

export default ThemeProviderSelector

const useTheme = router => {
  if (isBlogs(router.pathname)) {
    return blogTheme
  }

  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }

  if (isTet()) {
    return tetTheme
  }

  return theme
}

const selectGlobalCss = (router, theme) => {
  if (!isBlogs(router.pathname)) {
    return <HomeGlobalStyleCss theme={theme} />
  }
  return <BlogGlobalStyleCss theme={theme} />
}
