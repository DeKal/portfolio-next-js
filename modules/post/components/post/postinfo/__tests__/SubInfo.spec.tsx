import * as React from 'react'
import { shallowWithBaseTheme } from '~/modules/core/utils/withThemeProviders'
import SubInfo from '../SubInfo'
import theme from '~/modules/core/theme/styleguide/blogTheme'

describe('Post Info SubInfo', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithBaseTheme(<SubInfo />, theme)
    expect(wrap).toMatchSnapshot()
  })
})
