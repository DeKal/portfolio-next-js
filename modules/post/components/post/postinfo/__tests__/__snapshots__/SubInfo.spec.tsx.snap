// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Info SubInfo should render without throwing an error 1`] = `
<div
  className="SubInfo-sc-1bbx7ap-0 eHWfDZ"
/>
`;
